### PR TITLE
Fixing Typedoc breaking changes.

### PR DESCRIPTION
**Pull request by mbleigh**
Typedoc had a vulnerability, so I needed to update it which also necessitated updating TypeScript itself and working through a bunch of random breaking changes. I think I've got this properly wrangled now.

### CODE REVIEW

File: docgen/typedoc.js [hunk 1]
```diff
@@ -16,12 +16,12 @@
...
-  // includeDeclarations: true,
+  tsconfig: '../tsconfig.release.json',
   excludeExternals: true,
-  ignoreCompilerErrors: true,
+  exclude: ['../spec/**/*.ts', '../integration_test/**/*.ts'],
   name: 'Firebase Functions SDK',
-  mode: 'modules',
-  hideGenerator: true
+  hideGenerator: true,
+  // showConfig: true,
```
Comment (inlined) [inline docgen/typedoc.js:24]: Shouldn't we yank commented out lines?

**Review by inlined — APPROVED** [review]

File: docgen/typedoc.js (hunk 1 above)
Comment (mbleigh) [reply docgen/typedoc.js]: Sure.

**Comment by egilmorez** [comment]
As usual, appreciate your quick response on things Michael.  Thanks!

**Review by egilmorez — COMMENTED** [review]
Thanks!

**Pull request merged**